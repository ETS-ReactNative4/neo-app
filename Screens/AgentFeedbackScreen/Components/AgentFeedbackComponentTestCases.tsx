import React from "react";

import { ITestCase } from "../../../TypeInterfaces/TestCases/ITestCases";
import AgentInfoText from "./AgentInfoText";
import AgentFeedbackOption, {
  IAgentOptionData
} from "./AgentFeedbackOption/AgentFeedbackOption";

import AgentFeedback from "../AgentFeedback";

const agentOptionData: IAgentOptionData[] = [
  {
    text: "Superstar support",
    image: "https://i.imgur.com/YtdsUbs.png"
  },
  {
    text: "Timely response",
    image: "https://i.imgur.com/sYzOl65.png"
  },
  {
    text: "Empathy",
    image: "https://i.imgur.com/hm0u6k6.png"
  },
  {
    text: "Destination knowledge",
    image: "https://i.imgur.com/cd7irIa.png"
  }
];

const AgentFeedbackComponentTestCases: ITestCase[] = [
  {
    title: "Agent Info Text Section",
    Component: <AgentInfoText />
  },
  {
    title: "Agent Feedback Section",
    Component: <AgentFeedbackOption agentOptionData={agentOptionData} />
  },
  {
    title: "Agent Feedback Screen",
    Component: <AgentFeedback />
  }
];

export default AgentFeedbackComponentTestCases;
