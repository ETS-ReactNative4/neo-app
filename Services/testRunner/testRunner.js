import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
