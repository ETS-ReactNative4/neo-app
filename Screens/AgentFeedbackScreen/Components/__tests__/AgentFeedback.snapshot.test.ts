import AgentFeedbackComponentTestCases from "../AgentFeedbackComponentTestCases";
import testRunner from "../../../../Services/testRunner/testRunner";

AgentFeedbackComponentTestCases.map(testRunner);
