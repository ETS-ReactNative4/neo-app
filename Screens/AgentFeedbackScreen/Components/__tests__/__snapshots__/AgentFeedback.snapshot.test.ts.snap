// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agent Feedback Screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "https://i.imgur.com/zW6Eip0.png",
      }
    }
    style={
      Object {
        "height": 240,
        "left": 0,
        "position": "absolute",
        "resizeMode": "cover",
        "top": 0,
        "width": 240,
      }
    }
  />
  <View
    style={
      Object {
        "marginTop": undefined,
        "opacity": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 15,
          "lineHeight": 15,
          "marginBottom": 32,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Rate your experience with
    </Text>
  </View>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": undefined,
          },
        ],
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Image
        onError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://i.imgur.com/Uq2zUZA.png",
          }
        }
        style={
          Object {
            "height": 124,
            "marginBottom": 16,
            "width": 124,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "lineHeight": 16,
            "marginBottom": 4,
          }
        }
      >
        Mahesh Raja
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 15,
            "lineHeight": 15,
            "marginBottom": 32,
          }
        }
      >
        Your travel consultant
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(239, 239, 239, 1)",
            "height": 1,
            "marginBottom": 32,
            "width": 54,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": undefined,
        "transform": Array [
          Object {
            "scale": undefined,
          },
        ],
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 24,
            "width": 750,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "opacity": 0.1,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(44, 47, 66, 1)",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "voyager",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "opacity": 0.1,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(44, 47, 66, 1)",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "voyager",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "opacity": 0.1,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(44, 47, 66, 1)",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "voyager",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "opacity": 0.1,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(44, 47, 66, 1)",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "voyager",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "opacity": 0.1,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(44, 47, 66, 1)",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "voyager",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": undefined,
        "opacity": undefined,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "rgba(239, 239, 239, 1)",
            "borderTopWidth": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingVertical": 24,
            "width": 750,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 13,
            "lineHeight": 13,
            "marginBottom": 20,
          }
        }
      >
        What did you like the most?
      </Text>
      <View
        style={
          Object {
            "width": 750,
          }
        }
      >
        <ScrollView
          horizontal={true}
          showsHorizontalScrollIndicator={false}
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "width": 187.5,
                  }
                }
              >
                <Image
                  onError={[Function]}
                  source={
                    Object {
                      "uri": "https://i.imgur.com/YtdsUbs.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "borderRadius": 25,
                        "height": 56,
                        "marginBottom": 8,
                        "width": 56,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(119, 119, 119, 1)",
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 13,
                        "lineHeight": 15,
                        "paddingHorizontal": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Superstar support
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "width": 187.5,
                  }
                }
              >
                <Image
                  onError={[Function]}
                  source={
                    Object {
                      "uri": "https://i.imgur.com/sYzOl65.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "borderRadius": 25,
                        "height": 56,
                        "marginBottom": 8,
                        "width": 56,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(119, 119, 119, 1)",
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 13,
                        "lineHeight": 15,
                        "paddingHorizontal": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Timely response
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "width": 187.5,
                  }
                }
              >
                <Image
                  onError={[Function]}
                  source={
                    Object {
                      "uri": "https://i.imgur.com/hm0u6k6.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "borderRadius": 25,
                        "height": 56,
                        "marginBottom": 8,
                        "width": 56,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(119, 119, 119, 1)",
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 13,
                        "lineHeight": 15,
                        "paddingHorizontal": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Empathy
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "width": 187.5,
                  }
                }
              >
                <Image
                  onError={[Function]}
                  source={
                    Object {
                      "uri": "https://i.imgur.com/cd7irIa.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "borderRadius": 25,
                        "height": 56,
                        "marginBottom": 8,
                        "width": 56,
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(119, 119, 119, 1)",
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 13,
                        "lineHeight": 15,
                        "paddingHorizontal": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Destination knowledge
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </ScrollView>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "left": 32,
          "position": "absolute",
          "right": 32,
          "width": 686,
        },
        Object {
          "bottom": undefined,
          "opacity": undefined,
        },
      ]
    }
  >
    <View>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={[Function]}
        style={
          Object {
            "height": 56,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 56,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "rgba(0,198,132,1)",
                "borderRadius": 4,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(247, 247, 247)",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 17,
                  "lineHeight": 17,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`Agent Feedback Section 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "rgba(239, 239, 239, 1)",
        "borderTopWidth": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingVertical": 24,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgba(44, 47, 66, 1)",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 13,
        "lineHeight": 13,
        "marginBottom": 20,
      }
    }
  >
    What did you like the most?
  </Text>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <ScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "width": 187.5,
              }
            }
          >
            <Image
              onError={[Function]}
              source={
                Object {
                  "uri": "https://i.imgur.com/YtdsUbs.png",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 25,
                    "height": 56,
                    "marginBottom": 8,
                    "width": 56,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(119, 119, 119, 1)",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 13,
                    "lineHeight": 15,
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Superstar support
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "width": 187.5,
              }
            }
          >
            <Image
              onError={[Function]}
              source={
                Object {
                  "uri": "https://i.imgur.com/sYzOl65.png",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 25,
                    "height": 56,
                    "marginBottom": 8,
                    "width": 56,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(119, 119, 119, 1)",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 13,
                    "lineHeight": 15,
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Timely response
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "width": 187.5,
              }
            }
          >
            <Image
              onError={[Function]}
              source={
                Object {
                  "uri": "https://i.imgur.com/hm0u6k6.png",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 25,
                    "height": 56,
                    "marginBottom": 8,
                    "width": 56,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(119, 119, 119, 1)",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 13,
                    "lineHeight": 15,
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Empathy
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "width": 187.5,
              }
            }
          >
            <Image
              onError={[Function]}
              source={
                Object {
                  "uri": "https://i.imgur.com/cd7irIa.png",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 25,
                    "height": 56,
                    "marginBottom": 8,
                    "width": 56,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(119, 119, 119, 1)",
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 13,
                    "lineHeight": 15,
                    "paddingHorizontal": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Destination knowledge
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </ScrollView>
  </View>
</View>
`;

exports[`Agent Info Text Section 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onError={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.imgur.com/Uq2zUZA.png",
      }
    }
    style={
      Object {
        "height": 124,
        "marginBottom": 16,
        "width": 124,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgba(44, 47, 66, 1)",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 16,
        "lineHeight": 16,
        "marginBottom": 4,
      }
    }
  >
    Mahesh Raja
  </Text>
  <Text
    style={
      Object {
        "color": "rgba(83, 87, 109, 1)",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 15,
        "lineHeight": 15,
        "marginBottom": 32,
      }
    }
  >
    Your travel consultant
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(239, 239, 239, 1)",
        "height": 1,
        "marginBottom": 32,
        "width": 54,
      }
    }
  />
</View>
`;
